src/js/animator/core/AutoTake.js

```diff
@@ -0,0 +1,150 @@
+      AutoTake.enabled = true; // Is AutoTake is enabled?
+      AutoTake.isDark = false; // Is the current video frame dark (brightness below threshold)?
+      AutoTake.brightnessThreshold = 1.2;  // Brightness Threshold
+
+      // Add a list item to settings dialog
+      let autoTakeItem = document.createElement("li");
+      autoTakeList.appendChild(autoTakeItem);
+
+      // Item title
+      let itemTitle = document.createElement("div");
+      let itemTitleText = document.createElement("div");
+
+      let itemToggleBtn = document.createElement("div"); // Item toggle button
+
+      let itemSettingsContainer = document.createElement("div");
+      let brightnessThresholdInput = document.createElement("input"); // Brightness Threshold input
+      let brightnessThresholdText = document.createElement("div");
+
+      // Set title
+      autoTakeItem.appendChild(itemTitle);
+      itemTitle.classList.add("flex");
+      itemTitle.appendChild(itemTitleText);
+      itemTitleText.innerText = "Auto-Take";
+
+      // Add toggle button
+      itemTitle.appendChild(itemToggleBtn);
+      itemToggleBtn.setAttribute("style", "text-align: right");
+      itemToggleBtn.classList.add("auto-take-toggle-btn");
+      new ToggleButton(itemToggleBtn, function() {
+        if (AutoTake.enabled) {
+          clearInterval(AutoTake.autoTakeDaemon);
+          AutoTake.autoTakeDaemon = null;
+          AutoTake.enabled = false;
+        }
+        else {
+          // Start daemon to check webcam brightness 30 times per second. TODO: Make the interval time configurable
+          AutoTake.autoTakeDaemon = setInterval(AutoTake.checkBrightnessThreshold, 1000 / 30); // 30 FPS
+          AutoTake.enabled = true;
+        }
+        itemSettingsContainer.classList.toggle("hidden");
+      });
+
+      // Create item settings container to hold configurable inputs
+      autoTakeItem.appendChild(itemSettingsContainer);
+      itemSettingsContainer.classList.add("flex");
+      itemSettingsContainer.setAttribute("style", "align-items: center");
+
+      // Create brightness threshold text label and input
+      itemSettingsContainer.appendChild(brightnessThresholdText);
+      brightnessThresholdText.classList.add("flex");
+      brightnessThresholdText.setAttribute("style", "flex-basis: 80%");
+      brightnessThresholdText.innerText = "Brightness Threshold:";
+      itemSettingsContainer.appendChild(brightnessThresholdInput);
+      brightnessThresholdInput.style.margin = "0 0 0 0.5em";
+      brightnessThresholdInput.setAttribute("type", "number");
+      brightnessThresholdInput.setAttribute("min", "0");
+      brightnessThresholdInput.setAttribute("max", "255");
+      brightnessThresholdInput.setAttribute("step", "0.1");
+      brightnessThresholdInput.setAttribute("value", 1.2);
+
+      // Listen to the input being updated
+      brightnessThresholdInput.addEventListener("input", function(e) {
```

Comment:
This method should also correct users typing more than one decimal place

Reply:
Not really for this suggestion.
1. I feel users should able to specify the threshold to an arbitrary precision. The step size of 0.1 is merely for the up/down arrows on the input field, which I felt was reasonable for adjustment.
2. Auto-correcting to 1 decimal place feels very jarring and adds a lot of complexity.